tensor/genlib/stdeng_add.go

```diff
@@ -179,24 +179,32 @@ func add{{short .}}VS(a []{{asType .}}, b {{asType .}}) {
 
 func add{{short .}}Iter(a, b []{{asType .}}, ait, bit Iterator) (err error) {	
 	var i, j int
+	var validi, validj bool
 	for {
-		if i, _, err = ait.NextValid(); err != nil {
+		if i, validi, err = ait.NextValidity(); err != nil {

```

Comment:
I'm lost. What does `bit.NextValid(i)` do? Also what would happen when there are two slices with wildly differing strides?

For example:

```
T := New(WithShape(100, 200), Of(Float64)) // Dense (100,200)
S := T.Slice(S(0, 100, 20), S(0, 200, 2)) // T[0:100:20] - only every 20th row and every 2nd col
A := New(WithShape(5, 100), Of(Float64)) // Dense(5, 100)

// what would happen to : Add(A, S) 

bit := FlatIteratorFromDense(S)
// this would have valid j = [0, 2, 4, 6...] 
```